### ISSUE TITLE
Generating HDF5 files on Custom Dataset :

### ISSUE THREAD

**Issue by lakshaymehra** [frostinassiky/gtad#7]
Hi,

I am trying to use GTAD on my own dataset. I have annotated the data according to Thumos'14 annotation format.
I have used the MMAction repository to extract the RGB Frames as well. After extraction, I have a folder for each video containing all the RGB frames in that video as images.
However, I do not understand what my next step should be to train GTAD because GTAD requires a HDF5 file containing the RGB features of the dataset.

1. Can you please let me know how you generated the HDF5 containing I3D Features for the validation and test sets?
2. After generating the features, did you use TAG or did you use BSN to generate proposals? 

Your answers to the above questions will clarify a lot of doubts.

Thank you for your time!

**Comment by coolbay** [comment]
Hi, @lakshaymehra !

1. GTAD uses the TSN rgb-flow features in the HDF5 format as we provide in this [link](https://drive.google.com/drive/folders/10PGPMJ9JaTZ18uakPgl58nu7yuKo8M_k?usp=sharing). So what you can do is to first extract optical flow. Then extract features from both streams for your own dataset (TSN Or I3D as you wish) and store them as HDF5 files.  You can refer to [this](https://github.com/yjxiong/temporal-segment-networks) for extracting optical flow and TSN features. 
2. After generating features, we use our proposed G-TAD to generate proposals.

Thank you for your question!

**Comment by lakshaymehra** [comment]
Thank you for your detailed response. I will try the exact steps as mentioned by you. Really appreciate your time!

**Comment by lakshaymehra** [comment]
Hey @coolbay and @Frostinassiky,

I have generated the HDF5 file for my custom data set. The data/thumos_annotations folder in the repo contains separate CSV files for the val and test set of THUMOS'14. 

However, the code inside dataset.py requires a CSV file and a JSON file.

1. Do I only need to generate a  CSV file for my training data and a CSV file for my validation data?
Or do I also need to generate a JSON file?
2. If I need to generate a JSON file for a custom dataset, what are the contents of that? Do I also need to create separate JSON files for train and validation sets? Can you share an example of the same?

Thank you for your time!

**Comment by destousok** [comment]
Hey @coolbay and @lakshaymehra! I would like to ask that with referring to the extraction of features using TSN there is no detail about which layer of the network we should use as the output. Do you know which one is it?
Thank you! :)

**Comment by coolbay** [comment]
@destousok The second-to-last fc layer before predicting the categories.

**Comment by mrlihellohorld** [comment]
> Thank you for your detailed response. I will try the exact steps as mentioned by you. Really appreciate your time!

Hi， could u share the code for extracting feature?
